storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/serialization/CompressedBranchInfoSerializer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server.kvstore.serialization;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.ssz.tree.TreeNodeSource.CompressedBranchInfo;
+
+public class CompressedBranchInfoSerializer implements KvStoreSerializer<CompressedBranchInfo> {
+
+  @Override
+  public CompressedBranchInfo deserialize(final byte[] rawData) {
+    final Bytes data = Bytes.wrap(rawData);
+    checkArgument(
+        (data.size() - Integer.BYTES) % Bytes32.SIZE == 0,
+        "Branch data was an invalid length %s",
+        data);
+    // Take unsigned int from front (depth)
+    final int depth = data.getInt(0);
```

Comment:
nit: maybe we just say `take depth (int) from the front`

Reply:
Done.